core/src/main/java/hudson/FilePath.java

```diff
@@ -1271,7 +1271,7 @@ public void deleteRecursive() throws IOException, InterruptedException {
         private static final long serialVersionUID = 1L;
         @Override
         public Void invoke(File f, VirtualChannel channel) throws IOException {
-            deleteRecursive(deleting(f));
+            Util.deleteRecursive(deleting(f));
```

Comment:
The previous implementation of `deleteRecursive` would call `deleting(...)` on each file in the directory. Does the new `Util.deleteRecursive` still do this? If not, I think this might be a problem for agent-master security.

Reply:
Oh, great point! I'll fix this, thanks.